tests/integration/Zone/ZoneQueryRequestTest.php

```diff
@@ -3,104 +3,81 @@
-        $this->assertInstanceOf(Zone::class, $result->getAt(0));
-        $this->assertSame($zone->getId(), $result->getAt(0)->getId());
+        $client = $this->getApiClient();
+
+        ZoneFixture::withZone(
+            $client,
+            function (Zone $zone) use ($client) {
+                $request = RequestBuilder::of()->zones()->query()
+                    ->where('name=:name', ['name' => $zone->getName()]);
+                $response = $this->execute($client, $request);
+                $result = $request->mapFromResponse($response);
+
+                $this->assertCount(1, $result);
+                $this->assertInstanceOf(Zone::class, $result->current());
+                $this->assertSame($zone->getId(), $result->current()->getId());
+            }
+        );
     }
 
     public function testGetById()
     {
-        $draft = $this->getDraft();
-        $zone = $this->createZone($draft);
-
-        $request = ZoneByIdGetRequest::ofId($zone->getId());
-        $response = $request->executeWithClient($this->getClient());
-        $result = $request->mapResponse($response);
-
-        $this->assertInstanceOf(Zone::class, $zone);
-        $this->assertSame($zone->getId(), $result->getId());
+        $client = $this->getApiClient();
+
+        ZoneFixture::withZone(
+            $client,
+            function (Zone $zone) use ($client) {
+                $request = RequestBuilder::of()->zones()->getById($zone->getId());
+                $response = $this->execute($client, $request);
+                $result = $request->mapFromResponse($response);
+
+                $this->assertInstanceOf(Zone::class, $zone);
+                $this->assertSame($zone->getId(), $result->getId());
+            }
+        );
     }
 
     public function testGetByKey()
     {
-        $draft = $this->getDraft()->setKey('key-' . $this->getTestRun());
-        $zone = $this->createZone($draft);
-
-        $request = ZoneByKeyGetRequest::ofKey($zone->getKey());
-        $response = $request->executeWithClient($this->getClient());
-        $result = $request->mapResponse($response);
-
-        $this->assertInstanceOf(Zone::class, $zone);
-        $this->assertSame($zone->getKey(), $result->getKey());
+        $client = $this->getApiClient();
+
+        ZoneFixture::withZone(
+            $client,
+            function (Zone $zone) use ($client) {
+                $request = RequestBuilder::of()->zones()->getByKey($zone->getKey());
+                $response = $this->execute($client, $request);
+                $result = $request->mapFromResponse($response);
+
+                $this->assertInstanceOf(Zone::class, $zone);
+                $this->assertSame($zone->getId(), $result->getId());
+                $this->assertSame($zone->getKey(), $result->getKey());
+            }
+        );
     }
-
+    //TODO for Consistancy: should I change the place of this test? in other part it's in createRequestTest or in Update
     public function testDeleteByKey()
```

Comment:
for consistancy: should we change the place of this test? in other part of the code it's in createRequestTest or in Update

Reply:
Not really necessary to move it.